schema/validator.go

```diff
@@ -54,6 +55,28 @@ func (v Validator) Validate(src io.Reader) error {
 			"schema %s: unable to validate", v)
 	}
 
+	if v == MediaTypeManifest {
```

Comment:
@wking 
> ?  ‘oci-image-tool validate …’ calls both image.Validate(Layout) [1]
> and schema.MediaType(Manifest|ManifestList|Config).Validate [2].  I
> expect we should be calling the schema validation from the image
> validation functions, but either way, code in the image validation
> functions is getting called by ‘oci-image-tool validate …’.

It is some hard to do based on current code.
What you mentioned [1] is on package order main -> image -> schema;
[2] is main -> schema;
I think it had better to add code to schema package, and better to only one place to be called by up layer. For that I figure out the PR. I tried other way, but it seems to be some brittle.